third_party/blink/renderer/platform/wtf/type_traits.h

```diff
@@ -104,6 +106,45 @@ struct IsSubclassOfTemplateTypenameSizeTypename {
   static const bool value = sizeof(SubclassCheck(t_)) == sizeof(YesType);
 };
 
+template <typename T, typename = void>
+struct IsDeterministicHash : std::false_type {};
+
+template <typename T>
+struct IsDeterministicHash<T, std::enable_if_t<T::IsDeterministicHash>>
+    : std::true_type {};
+
+template <typename T>
+constexpr bool IsDeterministicHashV = IsDeterministicHash<T>::value;
+
+template <typename T>
+struct IsPointerType : std::false_type {};
+
+template <typename T>
+struct IsPointerType<T*> : std::true_type {};
+
+template <typename T>
+struct IsPointerType<std::unique_ptr<T>> : std::true_type {};
+
+template <typename T>
+struct IsPointerType<scoped_refptr<T>> : std::true_type {};
```

Comment:
A few more pointer types I can think of. Not sure if they matter all too much:

* `Persistent`
* `WeakPersistent`
* `WeakPtr` / `WeakPtrBase`
* `raw_ptr`

Reply:
Thank you! Added.